transactor/src/main/scala/io/mediachain/util/datastore/RocksDatastore.scala

```diff
@@ -0,0 +1,37 @@
+package io.mediachain.util.datastore
+
+import org.rocksdb._
+import io.mediachain.transactor.Types.{DataObject, Datastore, Reference}
+import io.mediachain.util.cbor.CborAST.{CString, CValue}
+import io.mediachain.hashing.MultiHash
+
+class RocksDatastore(path: String) extends Datastore with AutoCloseable {
+  import RocksDatastore.RocksReference
+
+  val db = RocksDB.open(path)
+
+  override def put(obj: DataObject): Reference = {
+    val bytes = obj.toCborBytes
+    val hash  = MultiHash.hashWithSHA256(bytes)
+    val ref   = RocksReference(hash.bytes)
+
+    db.put(ref.id, bytes)
+
+    ref
+  }
+
+  override def close: Unit = {
+    db.close()
+  }
+}
+
+object RocksDatastore {
+  class RocksReference(val id: Array[Byte]) extends Reference {
```

Comment:
Also, it would be nice to have a toString that prints the base58 or multihash of the array for debugging.